compiler/rustc_transmute/src/layout/mod.rs

```diff
@@ -30,33 +30,46 @@ impl fmt::Debug for Byte {
 }
 
 pub(crate) trait Def: Debug + Hash + Eq + PartialEq + Copy + Clone {}
-pub trait Ref: Debug + Hash + Eq + PartialEq + Copy + Clone {}
+pub trait Ref: Debug + Hash + Eq + PartialEq + Copy + Clone {
+    fn min_align(&self) -> usize {
+        1
+    }
+
+    fn is_mutable(&self) -> bool {
+        false
+    }
+}
 
 impl Def for ! {}
 impl Ref for ! {}
```

Comment:
```suggestion
impl Ref for ! {
    fn min_align(&self) -> usize {
        unreachable!()
    }

    fn is_mutable(&self) -> bool {
        unreachable!()
    }
}
```

Reply:
ah, thanks, yeah I wasn't sure what was the right thing to put here